pkgs/http_client_conformance_tests/lib/src/redirect_tests.dart

```diff
@@ -7,6 +7,7 @@ import 'package:http/http.dart';
 import 'package:stream_channel/stream_channel.dart';
 import 'package:test/test.dart';
 
+import 'redirect_server.dart' as server;
```

Comment:
I would expect these imports to cause problems on the web.

Reply:
I would expect that too - but it seems to work in practice - the CI passes and running `dart test -p chrome` on my workstation also works.

Reply:
We might have broken our safeguards against importing `dart:io` when compiling through the test runner. I best we haven't broken them in google3 though. This is not going to work internally.

Reply:
I filed https://github.com/dart-lang/test/issues/1748 to make it easier to catch these eagerly in the test runner.

Reply:
Do I need to care if this works in google3? It seems like it is useful that this actually works.

But, if we want to change it, these approaches come to mind:
1. Try to always use `spawnHybridUri` and try to deal with emulators as a special chase (as I wrote @aam above) - the downside is 
2. Only use `spawnHybridUri` for the web and solve the problem using conditional imports (I think that I'd need 3 files per server) like:

`redirect_server.dart` - existing file
`redirect_server_web.dart` - calls `spawnHybridUri`
`redirect_server_vm.dart` - calls `hybridMain` directly

`redirect_test.dart` would then do:

```dart
import 'redirect_server_vm.dart'
    if (dart.library.html) 'redirect_server_web.dart';
```

3: Remove browser support for the tests.

I'd go with (2) - it is verbose but it should work. Any thoughts?

Reply:
> Do I need to care if this works in google3?

Yes. We want to have `package:http` tests running in google3.

> 2\. Only use `spawnHybridUri` for the web and solve the problem using conditional imports (I think that I'd need 3 files per server)

I think this is the best approach. We could probably write a little script in Dart or bash to create a `_web.dart` and `_vm.dart` file for each `_server.dart` file in a directory.

@jakemac53 - can you think of any other patterns we could use?

Reply:
conditional imports is going to be the best way I think (and will work internally, with some special config)

Reply:
Done.